### PR TITLE
fixed use of time strings to allow both seconds and  second etc

### PR DESCRIPTION
**Pull request by ronanstokes-db**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers. 
If it fixes a bug or resolves a feature request, please provide a link to that issue.

## Types of changes

What types of changes does your code introduce to dbx?
_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Change to tutorials, tests or examples
- [ ] Non code change (readme, images or other non-code assets)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. 
If you're unsure about any of them, don't hesitate to ask. We're here to help! 
This is simply a reminder of what we are going to look for before merging your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Submission does not reduce code coverage numbers
- [ ] Submission does not increase alerts or messages from LGTM

## Further comments

minor fix

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -76,6 +76,19 @@ def testParseTimeInterval2a(self):
...
+        self.assertEqual(timedelta(hours=1, seconds=1), interval)
+
+        interval2 = parse_time_interval("1 hour, 10 milliseconds")
+        self.assertEqual(timedelta(hours=1, milliseconds=10), interval2)
+
+        interval3 = parse_time_interval("1 hour, 10 microseconds")
+        self.assertEqual(timedelta(hours=1, microseconds=10), interval3)
+
+        interval4 = parse_time_interval("1 year, 4 weeks")
+        self.assertEqual(timedelta(weeks=56), interval4)
```
Comment (renardeinside) [inline tests/test_utils.py:90]: from readability perspective it's better to parametrize this test as described [here](https://stackoverflow.com/questions/32899/how-do-you-generate-dynamic-parameterized-unit-tests-in-python) for instance

File: dbldatagen/utils.py [hunk 2]
```diff
@@ -196,9 +196,9 @@ def parse_time_interval(spec):
...
         elif time_type in ["minutes", "minute"]:
             minutes = time_value
-        elif time_type in ["seconds", "seconds"]:
+        elif time_type in ["seconds", "second"]:
             seconds = time_value
-        elif time_type in ["microseconds", "microseconds"]:
+        elif time_type in ["microseconds", "microsecond"]:
             microseconds = time_value
         elif time_type in ["milliseconds", "millisecond"]:
             milliseconds = time_value
```
Comment (renardeinside) [inline dbldatagen/utils.py:204]: huge chain of if/elif is usually a sign that you have an improperly typed input. These operations share same logic, but are repetative. 
You'll better write something like:
```
unit_measure = ["second", "millisecond", "microsecond"]
for unit in unit_measure:
    if some_input in unit, f"{unit}s":
        do_something_with_it(some_input)
```

**Review by renardeinside — CHANGES_REQUESTED** [review]

File: tests/test_utils.py (hunk 1 above)
Comment (ronanstokes-db) [reply tests/test_utils.py:90]: good suggestion

File: dbldatagen/utils.py (hunk 2 above)
Comment (ronanstokes-db) [reply dbldatagen/utils.py:204]: logic is not the same - different component of time being assigned in each case  - but agree on general principal

File: tests/test_utils.py (hunk 1 above)
Comment (ronanstokes-db) [reply tests/test_utils.py]: but requires migrating tests from unit test to pytest 

Currently tests use pytest in unittest compatibility mode so parameterize does not work. Dont want additional external lib dependency for minor change so simply rewriting as for next loop

Will use simple for next loop with array of tuples for readability

**Review by renardeinside — APPROVED** [review]
LGTM!

**Pull request merged**